modules/setting/storage.go

```diff
@@ -30,20 +30,28 @@ func (s *Storage) MapTo(v interface{}) error {
 	return nil
 }
 
-func getStorage(rootCfg ConfigProvider, name, typ string, targetSec *ini.Section) Storage {
-	const sectionName = "storage"
-	sec := rootCfg.Section(sectionName)
+const storageSectionName = "storage"
 
+func getStorageSection(rootCfg ConfigProvider) *ini.Section {
+	storageSec := rootCfg.Section(storageSectionName)
 	// Global Defaults
-	sec.Key("MINIO_ENDPOINT").MustString("localhost:9000")
-	sec.Key("MINIO_ACCESS_KEY_ID").MustString("")
-	sec.Key("MINIO_SECRET_ACCESS_KEY").MustString("")
-	sec.Key("MINIO_BUCKET").MustString("gitea")
-	sec.Key("MINIO_LOCATION").MustString("us-east-1")
-	sec.Key("MINIO_USE_SSL").MustBool(false)
-	sec.Key("MINIO_INSECURE_SKIP_VERIFY").MustBool(false)
-	sec.Key("MINIO_CHECKSUM_ALGORITHM").MustString("default")
+	storageSec.Key("MINIO_ENDPOINT").MustString("localhost:9000")
+	storageSec.Key("MINIO_ACCESS_KEY_ID").MustString("")
+	storageSec.Key("MINIO_SECRET_ACCESS_KEY").MustString("")
+	storageSec.Key("MINIO_BUCKET").MustString("gitea")
+	storageSec.Key("MINIO_LOCATION").MustString("us-east-1")
+	storageSec.Key("MINIO_USE_SSL").MustBool(false)
+	storageSec.Key("MINIO_INSECURE_SKIP_VERIFY").MustBool(false)
+	storageSec.Key("MINIO_CHECKSUM_ALGORITHM").MustString("default")
+	return storageSec
+}
 
+// getStorage will read storage configurations from 4 possible ways
+// 1 read configurations from [$name] if the item exist
+// 2 read configurations from [storage.$name] if the item exist
+// 3 read configurations from [storage.$typ] if the item exist
+// 4 read configurations from [storage] if the item exist
```

Comment:
```suggestion
// 1 read configurations from [$name] if the setting keys exist (eg: name="attachments")
// 2 read configurations from [storage.$name] if the keys exist
// 3 read configurations from [storage.$type] if the keys exist (eg: type="local" or "minio")
// 4 read configurations from [storage] if the keys exist
```


TODO: how do they overwrite? early steps overwrite later, or later steps overwrite early?